openai/api_requestor.py

```diff
@@ -666,7 +667,10 @@ def _interpret_response_line(
                 headers=rheaders,
             )
         try:
-            data = json.loads(rbody)
+            if self.response_format == ResponseFormat.JSON:
+                data = json.loads(rbody)
+            else:
+                data = rbody
```

Comment:
I checked and the Content-Type header on the response is always set correctly (at least for the audio transcription responses), so we can deduce whether to json-decode without any other code changes.
For the `json` and `verbose_json` response formats, the Content-Type is 'application/json', while for `text, `srt`, and `vtt`, the Content-Type is `'text/plain; charset=utf-8'`.

```suggestion
            if rheaders.get('Content-Type') == 'application/json':
                data = json.loads(rbody)
            else:
                data = rbody
```